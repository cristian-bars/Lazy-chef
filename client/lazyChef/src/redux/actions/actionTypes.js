export default {
  LOAD_RECIPES: 'LOAD_RECIPES',
  DELETE_RECIPE: 'DELETE_RECIPE',
  ADD_RECIPE: 'ADD_RECIPE',
  UPDATE_RECIPE: 'UPDATE_RECIPE',
  LOAD_RECIPE: 'LOAD_RECIPE',

  LOAD_USERS: 'LOAD_USERS',
  DELETE_USER: 'DELETE_USER',
  ADD_USER: 'ADD_USER',
  UPDATE_USER: 'UPDATE_USER',
  LOAD_USER: 'LOAD_USER',
};
