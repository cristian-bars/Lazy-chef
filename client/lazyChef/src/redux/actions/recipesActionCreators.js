import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/recipes';
const url = 'http://192.168.0.29:2022/recipes';
//const url = 'http://192.168.1.130:2022/recipes';

export function loadRecipes() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_RECIPES,
        recipes: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPES_ERROR',
      });
    }
  };
}

export function addRecipe(recipe) {
  return {
    type: actionTypes.ADD_RECIPE,
    recipe,
  };
}

export function deleteRecipe(recipeId) {
  return async dispatch => {
    await axios.delete(`${url}/${recipeId.id}`);
    dispatch({
      type: actionTypes.DELETE_RECIPE,
      recipeId,
    });
  };
}

export function updateRecipe(recipe) {
  return {
    type: actionTypes.UPDATE_RECIPE,
    recipe,
  };
}

export function loadRecipe(recipe) {
  return {
    type: actionTypes.LOAD_RECIPE,
    recipe,
  };
}

export function getRecipeById(recipeId) {
  console.log(`${url}/${recipeId}`);
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/${recipeId}`);
      dispatch({
        type: actionTypes.LOAD_RECIPE,
        recipe: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPE_ERROR',
      });
    }
  };
}
