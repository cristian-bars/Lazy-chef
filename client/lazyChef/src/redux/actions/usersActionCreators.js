import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/users';
const url = 'http://192.168.0.29:2022';
//const url = process.env.REACT_APP_PORT;

export function loadUsers() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_USERS,
        users: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}

export function addUsers(user) {
  return {
    type: actionTypes.ADD_USER,
    user,
  };
}

export function deleteUser(userId) {
  return async dispatch => {
    await axios.delete(`${url}/${userId.id}`);
    dispatch({
      type: actionTypes.DELETE_USER,
      userId,
    });
  };
}

export function updateUser(user) {
  return {
    type: actionTypes.UPDATE_USER,
    user,
  };
}

export function loadUser(user) {
  return {
    type: actionTypes.LOAD_USER,
    user,
  };
}

export function getUserById(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/login`, user);
      dispatch({
        type: actionTypes.LOAD_USER,
        userToken: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}
