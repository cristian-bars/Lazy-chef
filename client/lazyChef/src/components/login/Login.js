/* eslint-disable no-shadow */
import React, {useEffect, useState} from 'react';
import {connect} from 'react-redux';
import {View, Image, TouchableOpacity, Text, TextInput} from 'react-native';
import {getUserById} from '../../redux/actions/usersActionCreators';
import styles from './loginStyles';
import generalStyles from '../../../generalStyles';

const Login = ({navigation, dispatch, userAcces}) => {
  useEffect(() => {
    userAcces.token && navigation.navigate('Dashboard');
  }, [userAcces, navigation]);

  let [email, showEmail] = useState('');
  let [password, showPassword] = useState('');

  const loginClick = () => {
    dispatch(getUserById({email, password}));
  };

  return (
    <View style={generalStyles.container}>
      <Image
        style={generalStyles.logo}
        source={{
          uri: 'https://i.ibb.co/MVc000Y/Logo-transparent-Patxi.png',
        }}
      />
      <Text style={generalStyles.title} />
      <TextInput
        style={generalStyles.formLogin}
        testID="emailInput"
        placeholder="Email"
        value={email}
        onChangeText={email => showEmail(email)}
      />
      <TextInput
        style={generalStyles.formLogin}
        testID="passwordInput"
        placeholder="Password"
        value={password}
        onChangeText={password => showPassword(password)}
      />
      <TouchableOpacity
        style={generalStyles.button}
        testID="login"
        onPress={loginClick}>
        <Text style={generalStyles.baseText}>Acceder</Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.touchRegister}
        testID="register"
        onPress={() => navigation.navigate('Register')}>
        <Text style={styles.register}>Crear una cuenta</Text>
      </TouchableOpacity>
    </View>
  );
};

function mapStateToProps({userAcces}) {
  return {
    userAcces,
  };
}

export default connect(mapStateToProps)(Login);
