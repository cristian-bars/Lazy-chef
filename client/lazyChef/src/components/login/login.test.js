import 'react-native';
import React from 'react';
import Login from './index';
import thunk from 'redux-thunk';
import Store from 'redux-mock-store';
import * as actions from '../../redux/actions/usersActionCreators';
import {Provider} from 'react-redux';
import {render, fireEvent} from '@testing-library/react-native';

describe('Given a Login component', () => {
  const mockStore = Store([thunk]);
  let navigation;
  let myStore;
  beforeEach(() => {
    jest.spyOn(actions, 'getUserById').mockReturnValueOnce({type: ''});
    myStore = mockStore({
      userAcces: [
        {email: 'Cristian@gmail.com', password: '1a'},
        {email: 'Bars@gmail.com', password: '2a'},
      ],
    });
    navigation = {
      navigate: jest.fn(),
    };
  });

  test('renders correctly', () => {
    const wrapper = render(
      <Provider store={myStore}>
        <Login />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  describe('And user have token', () => {
    test('Then navigation is invoked ', () => {
      myStore = mockStore({
        userAcces: {
          token: '1234abc',
        },
      });

      render(
        <Provider store={myStore}>
          <Login navigation={navigation} />
        </Provider>,
      );

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });

  describe('When login button is pressed', () => {
    test('Then navigation.navigate is invoked', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <Login navigation={navigation} />
        </Provider>,
      );
      const login = getByTestId('login');
      fireEvent.press(login);

      expect(actions.getUserById).toHaveBeenCalled();
    });
  });

  describe('When emailInput is typing with cristian@gmail.com', () => {
    test('Then emailInput value is cristian@gmail.com', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <Login />
        </Provider>,
      );
      const emailInput = getByTestId('emailInput');
      const email = 'cristian@gmail.com';
      fireEvent.changeText(emailInput, email);

      expect(emailInput.props.value).toBe(email);
    });
  });

  describe('When passwordInput is typing with 1234ABC', () => {
    test('Then passwordInput value is 1234ABC', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <Login />
        </Provider>,
      );
      const passwordInput = getByTestId('passwordInput');
      const password = '1234ABC';
      fireEvent.changeText(passwordInput, password);

      expect(passwordInput.props.value).toBe(password);
    });
  });

  describe('When register button is pressed', () => {
    test('Then navigation.navigate is invoked', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <Login navigation={navigation} />
        </Provider>,
      );
      const register = getByTestId('register');
      fireEvent.press(register);

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
