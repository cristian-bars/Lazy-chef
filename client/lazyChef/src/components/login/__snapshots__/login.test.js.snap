// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Login component When is invoked Then it is renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#044AFD",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
      "paddingTop": 50,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://i.ibb.co/MVc000Y/Logo-transparent-Patxi.png",
      }
    }
    style={
      Object {
        "alignItems": "flex-start",
        "height": 250,
        "width": 250,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 25,
        "fontWeight": "bold",
        "paddingVertical": 20,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 5,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 5,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 30,
        "opacity": 1,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Cochin",
          "fontSize": 25,
        }
      }
    >
      Acceder
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Crear una cuenta
    </Text>
  </View>
</View>
`;

exports[`Given a Login component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#044AFD",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
      "paddingTop": 50,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://i.ibb.co/MVc000Y/Logo-transparent-Patxi.png",
      }
    }
    style={
      Object {
        "alignItems": "flex-start",
        "height": 250,
        "width": 250,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 25,
        "fontWeight": "bold",
        "paddingVertical": 20,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 5,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    testID="emailInput"
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 5,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    testID="passwordInput"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 30,
        "opacity": 1,
        "width": 300,
      }
    }
    testID="login"
  >
    <Text
      style={
        Object {
          "fontFamily": "Cochin",
          "fontSize": 25,
        }
      }
    >
      Acceder
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": 300,
      }
    }
    testID="register"
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Crear una cuenta
    </Text>
  </View>
</View>
`;

exports[`Login Form renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#044AFD",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
      "paddingTop": 50,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://i.ibb.co/MVc000Y/Logo-transparent-Patxi.png",
      }
    }
    style={
      Object {
        "alignItems": "flex-start",
        "height": 250,
        "width": 250,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 25,
        "fontWeight": "bold",
        "paddingVertical": 20,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 5,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 5,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 30,
        "opacity": 1,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Cochin",
          "fontSize": 25,
        }
      }
    >
      Acceder
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Crear una cuenta
    </Text>
  </View>
</View>
`;
